subsys/bluetooth/controller/ll_sw/ull_scan_aux.c

```diff
@@ -266,16 +354,30 @@ void ull_scan_aux_setup(memq_link_t *link, struct node_rx_hdr *rx)
 	}
 	aux->rx_last = rx;
 
-	lll->chan = aux_ptr->chan_idx;
-	lll->phy = BIT(aux_ptr->phy);
+	lll_aux->chan = aux_ptr->chan_idx;
+	lll_aux->phy = BIT(aux_ptr->phy);
+
+	/* See if this was already scheduled from LLL. If so, store aux context
+	 * in global scan struct so we can pick it when scanned node is received
+	 * with a valid context.
+	 */
+	if (is_lll_sched) {
+		LL_ASSERT(!lll->lll_aux);
 
-	aux_offset_us = ftr->radio_end_us - PKT_AC_US(pdu->len, phy);
+		lll->lll_aux = lll_aux;
+		lll_aux->state = 0U;
+
+		return;
+	}
+
```

Comment:
Here clear `lll->lll_aux`

```suggestion
	lll->lll_aux = NULL;
```